addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -425,7 +425,7 @@ def _get_invoice_line_price_vals(self, line):
         else:
             gross_price_subtotal = net_price_subtotal / (1.0 - (line.discount or 0.0) / 100.0)
         # Price subtotal with discount / quantity:
-        gross_price_unit = gross_price_subtotal / line.quantity if line.quantity else 0.0
+        gross_price_unit = gross_price_subtotal / line.quantity if line.quantity and gross_price_subtotal else 0.0
```

Comment:
I'm not sure you can do that. We're working with floats. 0.000000000000007 is not False in this kind of comparison while it represents 0.
You may need to use float_is_zero (or rather currency_id.is_zero(...))

(example: `a = 0.2 + 0.1; b = True if (a - 0.3) else False;` => `b==True`)